packages/relay/src/lib/eth.ts

```diff
@@ -1284,6 +1285,32 @@ export class EthImpl implements Eth {
         throw predefined.NON_EXISTING_ACCOUNT(call.from);
       }
     }
+
+    // verify gas is withing the allowed range
+    if (call.gas && call.gas > constants.BLOCK_GAS_LIMIT) {
+      throw predefined.GAS_LIMIT_TOO_HIGH(call.gas, constants.BLOCK_GAS_LIMIT);
+    }
+
+    // If "BlockParam" is distinct from blank, we check is a valid block number
+    if(blockParam) {
+      const blockNum = Number(blockParam);
```

Comment:
Could `blockParam` also be `{blockHash}`? [EIP-1898](https://eips.ethereum.org/EIPS/eip-1898) allows for `{blockHash}` to be passed instead of the `blockNumber` for various methods including `eth_call`.

So the type of `blockParam` should actually be extended to:
```ts
type BlockParam = {
  blockHash: string
} | string | null;
```

Reply:
Yes it could. Good point.
We have goals for this with https://github.com/hashgraph/hedera-json-rpc-relay/issues/822
In this implementation I'm covering our existing logic of just numbers.
Let me explore how to comment this as the above ticket would address this consideration